arm/Microsoft.DataFactory/factories/deploy.bicep

```diff
@@ -125,26 +131,22 @@ resource dataFactory 'Microsoft.DataFactory/factories@2018-06-01' = {
   }
 }
 
-resource dataFactory_managedVirtualNetwork 'Microsoft.DataFactory/factories/managedVirtualNetworks@2018-06-01' = {
-  parent: dataFactory
-  name: 'default'
-  properties: {}
+module dataFactory_managedVirtualNetwork 'managedVirtualNetwork/deploy.bicep' = if (!empty(managedVirtualNetworkName)) {
+  name: '${uniqueString(deployment().name, location)}-ManagedVirtualNetwork'
+  params: {
+    name: managedVirtualNetworkName
+    dataFactoryName: dataFactory.name
+  }
 }
 
-resource dataFactory_integrationRuntime 'Microsoft.DataFactory/factories/integrationRuntimes@2018-06-01' = {
-  parent: dataFactory
-  name: 'AutoResolveIntegrationRuntime'
-  properties: {
-    type: 'Managed'
-    managedVirtualNetwork: {
-      referenceName: 'default'
-      type: 'ManagedVirtualNetworkReference'
-    }
-    typeProperties: {
-      computeProperties: {
-        location: 'AutoResolve'
-      }
-    }
+module dataFactory_integrationRuntime 'integrationRuntime/deploy.bicep' = if (!empty(integrationRuntime)) {
+  name: '${uniqueString(deployment().name, location)}-IntegrationRuntime'
+  params: {
+    dataFactoryName: dataFactory.name
+    name: integrationRuntime.name
+    type: integrationRuntime.type
+    managedVirtualNetworkName: integrationRuntime.managedVirtualNetworkName
```

Comment:
As this value seems optional you should handle it as such: `managedVirtualNetworkName: contains(integrationRuntime, 'managedVirtualNetworkName') ? integrationRuntime.managedVirtualNetworkName : ''`

Reply:
Updated!

Reply:
Just one minor change: Please remove the ecapuslating brackets ( `(` ) they are redundant

Reply:
Done!